site/cds_rdm/inspire_harvester/writer.py

```diff
@@ -30,81 +32,126 @@ def _write_entry(self, entry, *args, **kwargs):
 
         existing_records_hits = existing_records.to_dict()["hits"]["hits"]
         existing_records_ids = [hit["id"] for hit in existing_records_hits]
+        current_app.logger.debug(
+            f"IDs of existing records found: {existing_records_ids}."
+        )
 
         if multiple_records_found:
+            current_app.logger.error(
+                f"{existing_records.total} records found on CDS with the same INSPIRE ID ({inspire_id})."
+            )
             raise WriterError(
                 f"More than 1 record found with INSPIRE id {inspire_id}. CDS records found: {', '.join(existing_records_ids)}"
             )
         elif should_update:
             self.update_record(entry, record_pid=existing_records_ids[0])
+            current_app.logger.info("Record has been successfully updated.")
         elif should_create:
             # no existing record in CDS - create and publish a new one
             self._create_new_record(entry)
+            current_app.logger.info("New record has been successfully created.")
         else:
             raise NotImplemented()
 
     def write(self, stream_entry, *args, **kwargs):
         """Creates or updates the record in CDS."""
         entry = stream_entry.entry
+        current_app.logger.info(
+            f"Writer entry point. Start processing stream entry: {entry}."
+        )
         self._write_entry(entry, *args, **kwargs)
+        current_app.logger.info(f"Writing entry finished.")
         return stream_entry
 
     def write_many(self, stream_entries, *args, **kwargs):
         """Creates or updates the record in CDS."""
         entries = [entry.entry for entry in stream_entries]
-
+        current_app.logger.info(
+            f"Writer entry point. Start processing a batch of stream entries: {entries}."
+        )
         for entry in entries:
             self._write_entry(entry, *args, *kwargs)
+        current_app.logger.info(f"Writing entries finished.")
         return stream_entries
 
     def _get_existing_records(self, inspire_id):
         """Find records that have already been harvested from INSPIRE."""
+        current_app.logger.info(
+            f"Start retrieving existing records by INSPIRE ID ({inspire_id})."
+        )
         # for now checking only by inspire id
         filters = [
             dsl.Q("term", **{"metadata.identifiers.scheme": "inspire"}),
             dsl.Q("term", **{"metadata.identifiers.identifier": inspire_id}),
         ]
         combined_filter = dsl.Q("bool", filter=filters)
+        current_app.logger.debug(
+            f"Filter for searching for existing records: {combined_filter}."
+        )
```

Comment:
not so relevant to report IMO, because the filter will change only if we change the code. so it will always be the same message